R/similarityMatrix.R

```diff
@@ -63,102 +71,61 @@
 #' axis(2, at = (yaxis - min(yaxis))/(max(yaxis) - min(yaxis)), labels = yaxis)
 #'
 similarityMatrix <- function(x,
-                            template.scaled,
-                            similarity.measure){
-
-  sliding.func <- switch(similarity.measure,
-                         "cov" = sliding_cov,
-                         "cor" = sliding_cor)
-
-  ## Outer lapply: iterate over pattern scales considered;
-  ## each lapply iteration fills one row of the output similarity matrix.
-  similarity.list <- lapply(template.scaled, function(template.scaled.i){
-
-    ## Inner lapply: iterate over, possibly, multiple patterns;
-    ## each lapply iteration returns a vector whose each element corresponds
-    ## to the highest value of similarity between signal \code{x} and
-    ## a short pattern
-    ## at a time point corresponding to this vector's element.
-    sliding.func.out0 <- lapply(template.scaled.i, function(template.scaled.ik){
-      do.call(sliding.func, list(long = x, short = template.scaled.ik))
-    })
-
-    c(do.call(pmax, sliding.func.out0), rep(NA, length(template.scaled.i[[1]]) - 1))
-  })
-
+                             template.scaled,
+                             similarity.measure) {
+  if (similarity.measure == "cov") {
+    similarity.list <- get_cov_similarities(x, template.scaled)
+  } else if (similarity.measure == "cor") {
+    similarity.list <- get_cor_similarities(x, template.scaled)
+  } else {
+    stop("Only 'cov' and 'cor' measures supported")
+  }
   ## rbind list elements (which are vectors) into a matrix
-  similarity.mat <- do.call(rbind, similarity.list)
-  return(similarity.mat)
-
+  similarity <- do.call(rbind, lapply(similarity.list, function(i) i[[1]]))
+  similarity_idx <- do.call(rbind, lapply(similarity.list, function(i) i[[2]]))
+  return(list("similarity" = similarity, "idx" = similarity_idx))
```

Comment:
Good point, i agree keeping interfaces consistent would be ideal. I don't have a good sense of the user base or how disruptive a change like this would be.

Will try and come up with some ideas for this. Perhaps some sort of `return_template_idx=FALSE` default argument that returns just the similarity matrix when false, but when true returns both?  (but then we're returning two different types based on what the flag is).